test.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-env jest */
+const Redux = require('redux')
+const awaitState = require('.')
+
+const initialState = { enabled: false }
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case 'TOGGLE_ENABLED':
+      return { ...state, enabled: !state.enabled }
+    case 'OTHER_ACTION':
+      return { ...state, someOtherProperty: 42 }
+    default:
+      return state
+  }
+}
+
+const isEnabledSelector = (store) => !!store.enabled
+const isDisabledSelector = (store) => !store.enabled
+
+let store
+
+beforeEach(() => {
+  store = Redux.createStore(reducer)
+})
+
+test('awaitState waits for state to update', async () => {
+  // queue up actions
+  setTimeout(() => store.dispatch({ type: 'TOGGLE_ENABLED' }), 1000)
```

Comment:
nit: do these need to be this long? Or maybe we can use fake timers?

Reply:
Probably not; they're just there to ensure the event loop clears; I don't know how short you could make them until you run into trouble. Also serve a secondary purpose, as you can now manually verify test run times look correct, to ensure it's actually waiting on the action.

Reply:
There's no point in using fake timers; the only purpose of timers is to delay stuff beyond the end of the event loop.

Reply:
the point is to make the tests run fast. Why not use nextTick if that's the only purpose?

Reply:
`nextTick` might work; I just don't know the internals of redux, if they're not fully synchronous, you no longer have guaranteed execution order.

Reply:
they are fully synchronous